package.json

```diff
@@ -54,9 +63,6 @@
     }
   },
   "lint-staged": {
-    "!(*.d).ts": [
-      "false -- Please use .tsx instead of .ts!"
-    ],
```

Comment:
Hmm… how about this?

```json
  "lint-staged": {
    "src/**/!(*.d).ts": [
      "false -- Please use .tsx instead of .ts in src/!"
    ],
    "*.js": [
      "false -- Please use .tsx with `// @ts-nocheck` instead of .js!"
    ],
    "*.{html,json,md}": [
      "prettier --write"
    ],
    "*.{ts,tsx}": [
      "bash -c tsc",
      "eslint --cache --fix",
      "prettier --write"
    ]
  }
```

I tested it and it correctly complains about `.ts` in `src/` only.